ethexe/consensus/src/tests.rs

```diff
@@ -0,0 +1,205 @@
+// This file is part of Gear.
+//
+// Copyright (C) 2025 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use std::vec;
+
+use crate::{BatchCommitmentValidationReply, BatchCommitmentValidationRequest};
+use ethexe_common::{
+    gear::{BlockCommitment, CodeCommitment, Message, StateTransition},
+    ProducerBlock, SimpleBlockData,
+};
+use ethexe_db::{BlockHeader, BlockMetaStorage, CodeInfo, CodesStorage, Database, OnChainStorage};
+use ethexe_signer::{Address, Digest, PrivateKey, PublicKey, SignedData, Signer};
+use gprimitives::H256;
+
+#[test]
+fn block_producer_index_calculates_correct_index() {
+    let validators_amount = 5;
+    let slot = 7;
+    let index = crate::block_producer_index(validators_amount, slot);
+    assert_eq!(index, 2);
+}
+
+pub fn init_signer_with_keys(amount: u8) -> (Signer, Vec<PrivateKey>, Vec<PublicKey>) {
```

Comment:
more likely mock module.
as well as its not a good practice to have some fns defined after tests

Reply:
rename tests to mock and  move one test from mock to lib.rs